tasks/task_manager.cc

```diff
@@ -17,6 +17,62 @@ namespace tasks {
 
 logging::logger tmlogger("task_manager");
 
+bool task_manager::task::children::all_finished() const noexcept {
+    return _children.empty();
+}
+
+size_t task_manager::task::children::size() const noexcept {
+    return _children.size() + _finished_children.size();
+}
+
+future<> task_manager::task::children::add_child(foreign_task_ptr task) {
+    rwlock::holder exclusive_holder = co_await _lock.hold_write_lock();
+
+    auto id = task->id();
+    const auto& [_, inserted] = _children.emplace(id, std::move(task));
```

Comment:
nit: iwhen we're interested only in the insertion status, we usually write this as: `auto inserted = _children.emplace(id, std::move(task)).second;`